File: scalaz3src/main/z3/java/Z3Config.java
package z3.java;

import z3.Z3Wrapper;
import z3.Pointer;

public class Z3Config extends Pointer {
    public Z3Config() {
        super(Z3Wrapper.mkConfig());
    }

    public void delete() {
        Z3Wrapper.delConfig(this.ptr);
        this.ptr = 0;
    }

    public Z3Config setParamValue(String paramID, String paramValue) {
        Z3Wrapper.setParamValue(this.ptr, paramID, paramValue);
        return this;
    }
}

File: scalaz3src/main/z3/scala/Z3Config.scala
package z3.scala

import z3.{Z3Wrapper,Pointer}

sealed class Z3Config(params: (String,Any)*) {
  val ptr : Long = Z3Wrapper.mkConfig()

  for((k,v) <- params) {
    Z3Wrapper.setParamValue(this.ptr, k, v.toString)
  }

  def delete() : Unit = {
    Z3Wrapper.delConfig(this.ptr)
  }

  def setParamValue(paramID: String, paramValue: String) : Z3Config = {
    Z3Wrapper.setParamValue(this.ptr, paramID, paramValue)
    this
  }
}
